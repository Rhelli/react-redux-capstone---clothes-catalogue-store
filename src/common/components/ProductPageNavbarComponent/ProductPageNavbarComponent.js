import React from 'react';
import styles from './ProductPageNavbarComponent.module.scss';

const ProductPageNavbarComponent = () => (
  <div className={styles.navbarContainer}>
    <div className={styles.navbarLink}>
      <a href="/">·êä Back</a>
    </div>
    <div className={styles.navbarTitle}>
      <h1>Clothes Inc.</h1>
    </div>
  </div>
);

export default ProductPageNavbarComponent;
