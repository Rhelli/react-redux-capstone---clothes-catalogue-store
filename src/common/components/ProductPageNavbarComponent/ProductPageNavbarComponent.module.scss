.navbarContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fdf2e9;
  height: 60px;
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  right: 0;
  border-bottom: 1px solid #ddd4cc;
  -webkit-box-shadow: 0 3px 6px -4px rgba(105, 105, 105, 1);
  -moz-box-shadow: 0 3px 6px -4px rgba(105, 105, 105, 1);
  box-shadow: 0 3px 6px -4px rgba(105, 105, 105, 1);
  z-index: 3;
}

.navbarTitle {
  font: 700 20px/20px 'Playfair Display', sans-serif;
  color: #303030;
  padding: 12px 16px;
}

.navbarLink {
  position: absolute;
  left: 10px;
  font: 400 14px/10px 'Source Sans Pro', monospace;

  a {
    text-decoration: none;
    background-color: #303030;
    border-radius: 25px;
    color: #ddd4cc;
    padding: 6px 10px;
    transition: 0.1s;

    &:hover {
      color: #fff;
      transition: 0.1s;
    }

    &:focus {
      outline: none;
      box-shadow: none;
    }
  }
}

@media only screen and (min-width: 768px) {
  .navbarContainer {
    height: 70px;
  }

  .navbarTitle {
    font-size: 22px;
    line-height: 22px;
  }

  .navbarLink {
    font-size: 18px;
  }
}

@media only screen and (min-width: 1200px) {
  .navbarTitle {
    font-size: 26px;
  }
}
