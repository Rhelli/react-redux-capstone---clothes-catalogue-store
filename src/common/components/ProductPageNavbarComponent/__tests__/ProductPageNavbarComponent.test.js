import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ProductPageNavbarComponent from '../ProductPageNavbarComponent';

describe('The appearance of the ProductPageNavbarComponent', () => {
  it('Matches the ProductPageNavbarComponent snapshot', () => {
    const snapshot = renderer.create(<ProductPageNavbarComponent />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Contains two main div containers', () => {
    const { container } = render(<ProductPageNavbarComponent />);
    expect(container.firstChild.childElementCount).toEqual(2);
  });

  it('Has an anchor link to the home page as its first element', () => {
    const { container } = render(<ProductPageNavbarComponent />);
    expect(container.firstChild.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.firstChild.classList.contains('navbarLink'));
    expect(container.firstChild.firstChild.firstChild.nodeName).toBe('A');
    expect(container.firstChild.firstChild.firstChild).toHaveAttribute('href', '/');
  });

  it('Has the brand text as the last element of the navbar', () => {
    const { container, getByText } = render(<ProductPageNavbarComponent />);
    expect(container.firstChild.lastChild.nodeName).toBe('DIV');
    expect(container.firstChild.lastChild.classList.contains('navbarTitle')).toBe(true);
    expect(container.firstChild.lastChild.firstChild.nodeName).toBe('H1');
    getByText('Clothes Inc.');
  });
});
