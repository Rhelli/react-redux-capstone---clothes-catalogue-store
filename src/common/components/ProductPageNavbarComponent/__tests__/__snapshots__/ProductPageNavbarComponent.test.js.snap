// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the ProductPageNavbarComponent Matches the ProductPageNavbarComponent snapshot 1`] = `
<div
  className="navbarContainer"
>
  <div
    className="navbarLink"
  >
    <a
      href="/"
    >
      ·êä Back
    </a>
  </div>
  <div
    className="navbarTitle"
  >
    <h1>
      Clothes Inc.
    </h1>
  </div>
</div>
`;
